setup_profile.py

```diff
@@ -0,0 +1,145 @@
+#! /usr/bin/env python3
+import yaml
+import os
+import shutil
+from pathlib import Path
+from typing import Union, Tuple
+
+import click
+from rdflib import Graph, FOAF, BNode, Literal
+
+from nanopub import NanopubClient, Nanopub
+from nanopub.definitions import USER_CONFIG_DIR
+from nanopub.namespaces import NPX, ORCID
+from nanopub.java_wrapper import JavaWrapper
+
+PRIVATE_KEY_FILE = 'id_rsa'
+PUBLIC_KEY_FILE = 'id_rsa.pub'
+DEFAULT_PRIVATE_KEY_PATH = USER_CONFIG_DIR / PRIVATE_KEY_FILE
+DEFAULT_PUBLIC_KEY_PATH = USER_CONFIG_DIR / PUBLIC_KEY_FILE
+PROFILE_PATH = USER_CONFIG_DIR / 'profile.yml'
+RSA = 'RSA'
+
+
+@click.command(help='Interactive CLI to create a nanopub user profile. A local version of the profile will be stored '
+                    'in the '
+                    'user config dir (by default HOMEDIR/.nanopub/). The profile will also be published to the '
+                    'nanopub servers.')
+@click.option('--keypair', nargs=2, type=Path,
+              prompt=f'If the public and private key you would like to use are not in {USER_CONFIG_DIR}, '
+                     f'provide them here. If they are in this directory or you wish to generate new keys, '
+                     f'leave empty.',
+              help='Your RSA public and private keys with which your nanopubs will be signed',
+              default=None)
+@click.option('--orcid', type=str, prompt=True, help='Your ORCID')
+@click.option('--name', type=str, prompt=True, help='Your name')
+@click.option('--publish/--no-publish', type=bool, is_flag=True, default=True,
+              help='If true, nanopub will be published to nanopub servers',
+              prompt=('Would you like to publish your profile to the nanopub servers?'
+                      'this links your ORCID to your RSA key, thereby making all your'
+                      'publications linkable to you'))
+def main(orcid, publish, name, keypair: Union[Tuple[Path, Path], None]):
+    """
+    Interactive CLI to create a user profile.
+
+    :param orcid: the users orcid or other form of universal identifier
+    :param publish: if True, profile will be published to nanopub servers
+    :param name: the name of the user
+    :param keypair: a tuple containing the paths to the public and private RSA key to be used to sign nanopubs. If
```

Comment:
We use google docstring formatting throughout the rest of the project, I'll change it.